packages/eslint-plugin-material-ui/src/rules/no-direct-state-access.js

```diff
@@ -22,7 +22,7 @@ function reportIfDirectlyAccessingState(node, context, nodeToReport = node) {
     return;
   }
 
-  const parserServices = ESLintUtils.getParserServices(context);
+  const { parserServices } = context.sourceCode;
```

Comment:
https://eslint.org/blog/2023/09/preparing-custom-rules-eslint-v9/

Reply:
@mui/xgrid Could you double-check if this rule is still correct and functioning?
I tried triggering it in the code, based on the `test` file examples, but I either lack context or something has changed and it no longer works in essence. 🤷

Reply:
I doubt it ever worked consistently 😅
The rule was added in https://github.com/mui/mui-x/pull/3521 in December 2021, but this line using the restricted syntax was added even before that:

<img width="624" src="https://github.com/mui/mui-x/assets/13808724/57eed891-aa37-4b97-9b15-2fb2748e6f03">


I think we are good to go 👍🏻

Reply:
Gotcha, thanks for confirming...
And I thought that I was going crazy trying to make that rule work... 🤷